### ISSUE TITLE
[New sniff] No hard-coded menu names in nav_menu functions

### ISSUE THREAD

**Issue by jrfnl** [WPTRT/WordPress-Coding-Standards#93]
#### Rule type:
Error / Warning

#### Rule:

> In the `(register|wp)_nav_menu()` function, the theme should only use a variable for the menu name. No hard-coded menu names allowed.

The rule should be clarified and added to the handbook before any action is taken on this issue.

#### Theme check file covering this rule:
Unknown
Loosely related: https://github.com/WordPress/theme-check/issues/142

#### Decision needed:
* Is this really a rule ?
* If so, should it be a warning or error ?

#### To do:
- [ ] Add the rule in the Theme Review handbook
- [ ] Create unit tests
- [ ] Create new sniff

**Comment by grappler** [comment]
This falls under 
> Avoid hard coding to modify content. Instead, use function parameters, filters and action hooks where appropriate.

This should be an error.

By using a hardcoded a hard coded menu name it would require the menu to have the exactly same name which does not make sense for a publically released theme.

**Comment by joyously** [comment]
I think you are mistaken. A theme names its menu locations, not the user. They should not have to be in variables.  The menu location is not "content" as mentioned in that rule you quoted.

**Comment by grappler** [comment]
> A theme names its menu locations, not the user.

You are right but we are not talking about the location but the menu name.

These are the two parameter which is causing the confusing. We want to restrict the first not the second.
- 'menu' (int|string|WP_Term) Desired menu. Accepts (matching in order) id, slug, name, menu object.
-  'theme_location' (string) Theme location to be used. Must be registered with register_nav_menu() in order to be selectable by the user.

I hope this makes sense.
